File: integration-tests/hibernate-orm-panache/src/main/java/io/quarkus/it/panache/GenericEntity.java
package io.quarkus.it.panache;

import jakarta.persistence.MappedSuperclass;

import io.quarkus.hibernate.orm.panache.PanacheEntity;

@MappedSuperclass
public abstract class GenericEntity<T> extends PanacheEntity {
    public T t;
    public T t2;

    public T getT2() {
        return t2;
    }

    public void setT2(T t2) {
        this.t2 = t2;
    }
}

File: integration-tests/hibernate-orm-panache-kotlin/src/main/kotlin/io/quarkus/it/panache/kotlin/GenericEntity.kt
package io.quarkus.it.panache.kotlin

import io.quarkus.hibernate.orm.panache.kotlin.PanacheEntity
import jakarta.persistence.MappedSuperclass

@MappedSuperclass
abstract class GenericEntity<T> : PanacheEntity() {
    var t: T? = null
    var t2: T? = null
}
